src/tests/test_parse.py

```diff
@@ -278,3 +279,30 @@ def test_magic_args(ip, line, expected):
     args = magic_args(sql_line, line)
 
     assert args.__dict__ == complete_with_defaults(expected)
+
+
+@pytest.mark.parametrize(
+    "query, expected",
+    [
+        ("SELECT * FROM table where x > :x", "SELECT * FROM table where x > :x"),
+        ("SELECT * FROM table where x > ':x'", "SELECT * FROM table where x > '\\:x'"),
+        ('SELECT * FROM table where x > ":x"', 'SELECT * FROM table where x > "\\:x"'),
+        (
+            "SELECT * FROM table where x > '':x''",
+            "SELECT * FROM table where x > ''\\:x''",
+        ),
+        (
+            'SELECT * FROM table where x > "":x""',
+            'SELECT * FROM table where x > ""\\:x""',
```

Comment:
<img width="736" alt="Screenshot 2023-07-30 at 7 35 24 PM" src="https://github.com/ploomber/jupysql/assets/29096135/518c24b6-31b0-4d93-8b46-974b45e0dd5d">

Is this invalid usecase? It looks similar to `"SELECT * FROM table where x > ':x'"` ?

Also, what about `"SELECT * FROM table where x > "':x'" "` ?

Reply:
yes, they're are invalid, named parameters should not be quoted, otherwise they're considered string literals. I added a new warning: if named_paramstyle is True, the query contains quoted params and there is a defined variable with the same name, I'm showing a warning telling the user that if they want to use parameters, they need to unquote them